backend/api/serializers/computetask.py

```diff
@@ -251,17 +277,41 @@ def create(self, validated_data):
         outputs = validated_data.pop("outputs")
 
         compute_task = super().create(validated_data)
+        input_kinds = {algo_input.identifier: algo_input.kind for algo_input in compute_task.algo.inputs.all()}
 
         for order, data_sample in enumerate(data_samples):
             TaskDataSamples.objects.create(compute_task=compute_task, data_sample=data_sample, order=order)
 
         for position, input in enumerate(inputs):
-            ComputeTaskInput.objects.create(
+            task_input = ComputeTaskInput.objects.create(
                 channel=compute_task.channel,
                 task=compute_task,
                 position=position,
                 **input,
             )
+            # task input asset could be known during task registration
+            # or could be resolved later if it does not exist yet
+            if task_input.asset_key:
+                ComputeTaskInputAsset.objects.create(
+                    channel=compute_task.channel,
+                    task_input=task_input,
+                    asset_key=task_input.asset_key,
+                    asset_kind=input_kinds[task_input.identifier],
+                )
+            elif task_input.parent_task_key:
+                task_output = ComputeTaskOutput.objects.get(
+                    task=task_input.parent_task_key,
+                    identifier=task_input.parent_task_output_identifier,
+                )
+                # FIXME: we should use the position instead of the first asset
```

Comment:
We do not support multiple output assets yet, and the final implementation might not rely on positions.
IMO the comment should read:
```suggestion
                # This only supports a single asset per output for now
```